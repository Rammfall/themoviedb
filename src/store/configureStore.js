import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
