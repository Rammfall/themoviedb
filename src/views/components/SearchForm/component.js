import React from 'react'
import { Col, Row } from 'antd'
import { CloseOutlined } from '@ant-design/icons'
import { Form, Field } from 'formik'
import PropTypes from 'prop-types'

import InputField from 'Views/components/shared/InputField'

const SearchFormComponent = ({ handleSubmit, isLoading, handleReset }) => (
  <Row
    justify='center'
  >
    <Col
      className='gutter-row'
      xs={{ span: 20 }}
      sm={{ span: 20 }}
      md={{ span: 14 }}
      lg={{ span: 12 }}
      xl={{ span: 10 }}
    >
      <Form>
        <Field
          name='search'
          component={InputField}
          type='search'
          placeholder='Enter movie name'
          size='large'
          enterButton='Search'
          className='top-margin'
          prefix={<CloseOutlined onClick={handleReset} />}
          onSearch={handleSubmit}
          loading={isLoading}
        />
      </Form>
    </Col>
  </Row>
)

SearchFormComponent.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired,
  isLoading: PropTypes.bool
}

SearchFormComponent.defaultProps = {
  isLoading: false
}

export default SearchFormComponent
