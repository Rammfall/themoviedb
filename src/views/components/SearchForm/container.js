import React from 'react'
import { withFormik } from 'formik'
import { connect } from 'react-redux'
import { compose } from 'ramda'
import * as yup from 'yup'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import queryParams from 'Utils/router/queryParams'
import searchSubmit from 'Utils/formikEvents/searchSubmit'
import { loadingSelector } from 'Store/concepts/data/selectors'
import { dashboardConstant } from 'Constants/concepts'

import SearchFormComponent from './component'

const SearchForm = ({ handleSubmit, isLoading, handleReset }) => (
  <SearchFormComponent
    handleSubmit={handleSubmit}
    isLoading={isLoading}
    handleReset={handleReset}
  />
)

SearchForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired,
  isLoading: PropTypes.bool
}

SearchForm.defaultProps = {
  isLoading: false
}

const mapStateToProps = (state) => ({
  isLoading: loadingSelector(state, dashboardConstant)
})

export { SearchForm as SearchFormContainer }
export default compose(
  withRouter,
  connect(mapStateToProps),
  withFormik({
    validationSchema: yup.object().shape({
      search: yup.string()
    }),
    mapPropsToValues: ({ location: { search: query } }) => {
      const searchQuery = decodeURIComponent(queryParams(query).get('search') || '')

      return {
        search: searchQuery
      }
    },
    handleSubmit: searchSubmit
  })
)(SearchForm)
