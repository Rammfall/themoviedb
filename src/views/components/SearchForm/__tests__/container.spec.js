import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import SearchForm, { SearchFormContainer } from '../container'

jest.mock('react-router-dom', () => ({
  withRouter: component => component
}))
jest.mock('Utils/router/pushParam', () => jest.fn)

describe('SearchForm', () => {
  const store = configureStore()()

  it('matches snapshot', () => {
    const props = {
      store,
      location: {
        search: '?search='
      }
    }
    const wrapper = shallow(<SearchForm {...props} />)
    const container = diveTo(wrapper, SearchFormContainer)

    expect(container).toMatchSnapshot()
  })
})
