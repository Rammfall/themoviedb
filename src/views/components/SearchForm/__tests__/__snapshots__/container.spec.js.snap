// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm matches snapshot 1`] = `
<SearchFormComponent
  handleReset={[Function]}
  handleSubmit={[Function]}
  isLoading={false}
/>
`;
